Send Intent URL Normalization

Jacek-Gawel opened the pull request:
In cases where the URL sent in the intent contained spaces, the application did not open the given URL but instead used it as a search query in Google.

By normalizing the URL, the text from the intent is now correctly opened as a URL.

This issue is known to occur with Firebase App Distribution, which sends the app name in the intent. If the app name contains spaces, the link is not opened but is treated as a search query.

Replace Intent `ExtraText` spaces with `%20`

Jacek-Gawel committed:
Add url unification:

 - replace spaces with %20

plateaukao commented:
Thank you for providing the PR. I'll check whether it will make other scenarios malfunctioning; if not, I'll merge it.

plateaukao commented:
Hi, @Jacek-Gawel 
could you provide me several example of Firebase App Distribution usage?
I would like to verify the scenarios.
Also, could you fix indent issues in the PR?
Otherwise I may just send a commit directly based on your PR, in order not to send another PR to fix those indentations.

plateaukao committed:
ui: fix divider color in gpt query list

plateaukao committed:
feat: select context as a whole sentence

plateaukao committed:
feat: select text in a complete sentence

plateaukao committed:
feat: now text selection has a new feature to select whole sentence

plateaukao committed:
fix: sometime select sentence is not working due to simulating click point is not good enough

plateaukao committed:
feat: support individual model setting for each gpt action

plateaukao committed:
ver: v11.14.0

plateaukao committed:
fix: make sure when model or type is changed for gpt action, it will be applied immediately

plateaukao committed:
feat: support long press on action mode menu gpt actions to edit them

plateaukao committed:
feat: add font boldness dialog for applying font boldness

plateaukao committed:
feat: add select paragraph text, finetune UI for highlights Activity

plateaukao committed:
feat: move all per site settings to database table

plateaukao committed:
fix: db migration issue

plateaukao committed:
feat: make invert color a property for each site

plateaukao committed:
feat: support export feature for gpt queries

plateaukao committed:
feat: support other translation results to be saved, enhance the implementation of toast

plateaukao committed:
feat: support using volume key up down to scroll qpt queries

plateaukao committed:
feat: use md-block to display saved gpt queries html

plateaukao committed:
fix: issue with invert color in non-inverted url

plateaukao committed:
feat: enhance the logic of modifying translation configuration dialog

plateaukao committed:
feat: support in-app language setting

plateaukao committed:
string: update korean locale

plateaukao committed:
ver: v11.15.0

plateaukao committed:
long press bookmark sync to configure it, add ollama, gemini icons

plateaukao committed:
Update README.md

plateaukao committed:
refactor: move out view manipulation from actionModeViewModel

plateaukao committed:
feat: support bookmark reorder now!

plateaukao committed:
ver: update library versions

plateaukao committed:
refactor: remove redundant bookmark order activity, use existing UI to do so

plateaukao committed:
refactor: only drag handler, can drag without long pressing. Add different icon when drag mode is on.

plateaukao committed:
feat: add a toast to describe the dragging of bookmarks start working.

plateaukao committed:
feat: add a time info on toolbar.

plateaukao committed:
fix: order field is not written to exported bookmark.txt

plateaukao committed:
fix: when there are multiple saved tabs, the translation configuration will be messed up and show wrong translation to different tabs.

yoalerobledo committed:
Fix typo under setting_title_enable_inplace_translate

plateaukao committed:
fix: fix bookmark text press range

plateaukao committed:
Merge pull request #397 from yoaroib/main

Fix typo under setting_title_enable_inplace_translate

XuBaocai committed:
Update strings.xml

XuBaocai committed:
Update strings.xml

plateaukao committed:
fix: #400, #402

plateaukao committed:
Merge pull request #399 from XuBaocai/patch-21

Update strings.xml

plateaukao committed:
update build time, add url to history

plateaukao committed:
fix: #398 add an option to disable touch area long press actions

plateaukao committed:
ui: finetune history list, use shorter date string

plateaukao committed:
ui: fix issues when using custom font, facebook will show tofu characters

plateaukao committed:
ui: remove icons in settings, and use switch instead of check for toggling features

plateaukao committed:
ui: make statusbar white background

plateaukao committed:
fix: use transcation when re-ordering bookmarks

plateaukao committed:
ver: to 11.16.0

plateaukao committed:
ui: finetune switch toggle layout

plateaukao committed:
ui: fix translation label display logic

plateaukao committed:
feat: add spacer in toolbar configuration

plateaukao committed:
ui: reduce toolbar recompose frequency

plateaukao committed:
fix: first time action mode dialog position wrong.

plateaukao committed:
refactor: reduce translation dialog repaint frequency

plateaukao committed:
feat: support open other epub files

plateaukao committed:
fix: do not show bookmark sync dialog in edit dialog

plateaukao committed:
refactor: move ttsManager into ttsViewModel

plateaukao committed:
refator: move more functions out of BrowserActivity

plateaukao committed:
refactor: use application context and remove some redundant context argument

plateaukao committed:
fix: extract toolbar layout change codes from browser activity

plateaukao committed:
refactor: extract translation viewmodel codes

plateaukao committed:
refactor: actionmode visibility

plateaukao committed:
fix: input address bar is not focused at first time

plateaukao committed:
feat: add an behavior setting to enable using web cache first

plateaukao committed:
feat: when long clicking on setting icon on menu, go to settings

plateaukao committed:
fix: make sure only a small queue is used to fetch tts audio from openai

plateaukao committed:
fix: tts play speed, long press on setting

plateaukao committed:
ver: v11.17.0

plateaukao committed:
ui: remove un-used icons

plateaukao committed:
feat: make touch area actions configureable

plateaukao committed:
ui: fix string

plateaukao committed:
feat: add a close button to settings

plateaukao committed:
feat: support e tts

plateaukao committed:
feat: support changing voices for e tts

plateaukao committed:
ui: enhance voice display UI

plateaukao committed:
fix: useOpenAiTts and ttsType relation

plateaukao committed:
fix: color issue on tts voice dialogfragment

plateaukao committed:
refactor: remove un-used codes

plateaukao committed:
feat: add recently used voices

plateaukao committed:
fix: crash when two lanagues are mixed in contenxt for tts

plateaukao committed:
feat: new voices data

plateaukao committed:
refactor: bookmark loading way

plateaukao committed:
feat: move recent voices to tts dialog instead

plateaukao committed:
refactor: tts data format

plateaukao committed:
refactor: tts implementation

plateaukao committed:
refactor: merge etts files

wanderor committed:
Overwrite img src with data-src

Many website (e.g. Wechat) sets a placeholder for img src and loads the actual URL lazily. Mozilla Readability attempted to recognize and handle it, but the logic is narrow and does not cover many cases. Instead, this code change aggressively assume that we can overwrite img src with data-src, etc.

wanderor committed:
Log console.log()

wanderor committed:
Merge branch 'plateaukao:main' into main

plateaukao committed:
feat: make sure tts won't request too short strings.

plateaukao committed:
feat: support pause feature in tts setting dialog

plateaukao committed:
feat: add add article to read list

plateaukao committed:
fix: clear the to-be-read list after user click on stop

plateaukao committed:
feat: add support for adding web content to reading list

plateaukao committed:
feat: support read article in context menu. no need to open a link now

plateaukao committed:
feat: refactor codes, and prevent sentences with float numbers are separated.

plateaukao committed:
ui: add strings for tts types

plateaukao committed:
ver: 12.0.0 \n use real default voiceitem

plateaukao committed:
feat: use bytearray instead of file for voice reading to prevent from writing files

plateaukao committed:
ui: remove redundant tts type dialog

plateaukao committed:
feat: add openai voice options

plateaukao committed:
fix: pause play state

plateaukao committed:
fix: clean headless tts webview after loading text

plateaukao committed:
fix: try to fix duplicate read article requests due to onpagefinished logic

plateaukao committed:
fix: tts play button theming, toast for adding article to read list

plateaukao committed:
fix: use more precise word count by using regular expression

plateaukao committed:
fix: better getWordCount logic

wanderor committed:
Merge branch 'plateaukao:main' into main

plateaukao committed:
use reading state and speaking state to distinguish if reader is onprogress, and whether its paused or not

wanderor committed:
Reset reader mode properly

Currently, NinjaWebView does not reset reader mode properly. Apparently, reset logics are inconsistent across various methods. This code change introduces resetState() method to solve that.

I'm thinking that other fields should be included in resetState() as well: settings.textZoom, settings.cacheMode, isTranslateByParagraph. I'm not familiar with this codebase, so l'm leaving this for owner to decide.

plateaukao committed:
Merge pull request #408 from wanderor/main

Overwrite img src with data-src

plateaukao committed:
ui: add line-height: 1.5 to text translation

plateaukao committed:
Merge pull request #411 from wanderor/fix1

Reset reader mode properly

plateaukao committed:
ui: finetune setting UI

plateaukao committed:
feat: add play button into tts setting dialog

plateaukao committed:
ui: update icons on menu dialog

plateaukao committed:
refactor: remove un-used icons

plateaukao committed:
fix: websocket failure may stop tts functioning

plateaukao committed:
ui: change most toolbar default icon to Icons

plateaukao committed:
ui: move ui language setting to first

plateaukao committed:
feat: add fastforward to next article, but not working yet

plateaukao committed:
refactor: resetState of ninjaWebView

plateaukao committed:
fix: #412 fix popup menu not vertical scrollable, fix toolbar scrollling issue: scrollable when without spacers

plateaukao committed:
feat: support showing current reading content in tts dialog

plateaukao committed:
feat: now tts dialog won't be dismissed without pressing dismiss button or touch outside of dialog

plateaukao committed:
ui: use Icons in context menu dialog too

plateaukao committed:
ci: update upload script to v4

plateaukao committed:
feat: suppor translation in reading context too. need to press on current reading text

wanderor committed:
Bug fix: crash at startup

Error log:

2024-09-28 22:16:43.359  2119-2261  AndroidRuntime          info.plateaukao.einkbro              E  FATAL EXCEPTION: DefaultDispatcher-worker-2
                                                                                                    Process: info.plateaukao.einkbro, PID: 2119
                                                                                                    java.lang.NullPointerException: Attempt to invoke interface method 'boolean java.util.List.addAll(java.util.Collection)' on a null object reference
                                                                                                    	at info.plateaukao.einkbro.database.BookmarkManager$1.invokeSuspend(BookmarkDao.kt:227)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
                                                                                                    	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
                                                                                                    	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
                                                                                                    	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@9a91a3d, Dispatchers.IO]

plateaukao committed:
Merge pull request #414 from wanderor/main

Bug fix: crash at startup

plateaukao committed:
fix: sometimes tts text id empty

plateaukao committed:
ci: update ci toolchain

plateaukao committed:
refactor: remove un-used codes

plateaukao committed:
refactor

plateaukao committed:
ver: to 12.1.0

plateaukao committed:
fix: tts logic about ChannelData

plateaukao committed:
ui: change icon painterResource to ImageVector.vectorResource

plateaukao committed:
feat: add a crash handler to write logs in Download folder

plateaukao committed:
fix: build error

plateaukao committed:
Update bug-report-only.md

plateaukao committed:
fix: llm model is wrong

plateaukao committed:
feat: support fastforward to next article

plateaukao committed:
ui: use skip next icon

alchem1no committed:
Updated and rewrited language files

- Added translation for new strings;
- Strings in Russian and English language files now match;
- Rewritten some string in both English and Russian language files;
- Fixed Typos;
- Added comments to context dependent or difficult cases.

Although Russian file is mostly translated it still has some strings I haven't translated for various reasons: some of them aren't translated on purpose, some aren't because I didn't understand what it does, some are because I didn't find them in EinkBro and I don't understand in what context does it appear.

alchem1no committed:
Update Russian strings.xml

alchem1no committed:
Update English strings.xml

plateaukao committed:
Merge pull request #417 from Myron472/main

Updated and rewrited language files

plateaukao committed:
fix: translation text in tts is shown too late

plateaukao committed:
string: fix build error

alchem1no committed:
Language Improvements

- Made dot behavior more consistent by removing ones at the end of strings;
- Improved Russian translation;
- Changed default bookmarks file name that is shown when exporting bookmarks from *bookmark.txt* to *bookmarks.json*.

alchem1no committed:
Translation Improvements #2

- Added new strings to Russian strings file;
- Fixed words that should be written with capital letters.

alchem1no committed:
Fixed typo in ru strings

alchem1no committed:
Fix typo in ru strings

plateaukao committed:
Merge pull request #419 from Myron472/main

Language Improvements

plateaukao committed:
fix: string comma issue, finetune action menu label single line alignment

plateaukao committed:
ui: use Icons more

plateaukao committed:
ui: remove more xml icons

alchem1no committed:
Replace three dots with ellipsis

plateaukao committed:
Merge pull request #420 from Myron472/main

Replace three dots with ellipsis

plateaukao committed:
string: add back correct string labels

plateaukao committed:
fix: do not turn on web cache by default, because it may cause issues sometimes

plateaukao committed:
ui: re-order default toolbar actions

plateaukao committed:
fix: re-order default toolbar actions for phone

plateaukao committed:
ui: fix icon color

alchem1no committed:
Language: fix typos and term unconsistencies, rewrite some strings

alchem1no committed:
Fix: undercase letter

alchem1no committed:
Translation: Remove unnecessary capital letters, fix typos, rewrite some values | Fix: add line break symbol between two messages in whitelist menu

alchem1no committed:
Fix: uncomment string

alchem1no committed:
Fix: typo in comment

alchem1no committed:
Remove irrelevant comment

plateaukao committed:
Merge pull request #423 from Myron472/language

Language: replace capital letters, rewrite values, fix missing line break

plateaukao committed:
ci: make it build faster

plateaukao committed:
ver: v12.2.0

alchem1no committed:
Update values/strings.xml (#426)

plateaukao committed:
feat: support different styles for immersive translation

plateaukao committed:
ui: fine tune translation text style

alchem1no committed:
Update Gradle version (#427)

plateaukao committed:
ui: prevent translation text block too wide

plateaukao committed:
fix: crashes when there's no network

plateaukao committed:
ui: fix ui issue for paragraph translation

plateaukao committed:
refactor: extract javascript strings

plateaukao committed:
fix #384 fix launch from boox floating button issue. more scenarios: when search in same tab is not enabled in setting

plateaukao committed:
refactor: ninjawebview javascript loading logic

plateaukao committed:
refactor: more javascripts

plateaukao committed:
refactor: move more javascripts to asset files

alchem1no committed:
Update Gradle plugin (AGP) version (#432)

plateaukao committed:
refactor: extract more javascripts

plateaukao committed:
feat: remove ads from x

plateaukao committed:
refactor: web post processing

plateaukao committed:
refactor: rename NinjaWebView to EBWebView

plateaukao committed:
fix: remove deepl translatioin

plateaukao committed:
feat: remove highlight by clicking on the original highlight text selection

plateaukao committed:
refactor: remove redundant codes

plateaukao committed:
fix: bookmark deletion not reflected directly on bookmark list

plateaukao committed:
fix: bookmark edit not reflected issue

plateaukao committed:
fix: prune webtitle when saving bookmark or epub chapter name

plateaukao committed:
ver: to 12.3.0

plateaukao committed:
fix: crash in layout xml

plateaukao committed:
ui: re-write toolbar configuration UI

plateaukao committed:
ui: remove added actions section in new toolbar config

plateaukao committed:
ui: fix dark mode color

plateaukao committed:
ui: fix toolbar configuration spacer width issue!!

plateaukao committed:
ui: move toolbar config to bottom

plateaukao committed:
ui: finetune tollconfig ui in different resolution

plateaukao committed:
feat: modify url links

plateaukao committed:
ui: now the web title element size is prefectly calculated in toolbar

plateaukao committed:
ui: now title width is perfect too

plateaukao committed:
remove dependency of apache commons text

plateaukao committed:
feat: make sure new adblock works

plateaukao committed:
feat: add setting screen

plateaukao committed:
refactor: remove livedata variables

plateaukao committed:
feat: now filter download basically works, but ui is not up to date..

plateaukao committed:
feat: add post notification permission

plateaukao committed:
refactor: make Filter immutable, whenever it's changed, create new one

plateaukao committed:
Update README.md

change snapshot binary name, make default to arm64-v8

plateaukao committed:
feat: add save epub, share link function to toolbar

plateaukao committed:
feat: use web title as file name and book name

plateaukao committed:
feat: add invert color to toolbar

plateaukao committed:
fix: download snapshot binary with arm64-v8a cpu model

plateaukao committed:
fix: #440 when second webview is displayed and text selected, need to use selected text from it

plateaukao committed:
ver: to 14.0.0

plateaukao committed:
fix: now readaloud works in read epub activity too

plateaukao committed:
feat: you can select a text, and start tts read from that sentence

plateaukao committed:
fix: buildconfig issue

plateaukao committed:
ui: show spacer dashed border for easier configuration

plateaukao committed:
fix: #441 do not update to prelease version for Update to Latest Release button

plateaukao committed:
refactor: remove old adblock codes

plateaukao committed:
fix: temporarily disable scriplet to check what it's done and performance impact

plateaukao committed:
ui: draw border on top of tab list so that it's more clear

plateaukao committed:
ui: fix dark mode icon color on actionmode menu

plateaukao committed:
fix: incognito may impace save history value after toggling on and off

plateaukao committed:
ui: when in vertical reader mode, make sure the line interval is large enough

plateaukao committed:
ui: fine tune ui for vertical reader mode

plateaukao committed:
ui: fine tune ui for vertical reader mode

plateaukao committed:
fix: toolbar config for landscape

plateaukao committed:
fix: vertical read, starts from beginning

plateaukao committed:
ui: now text will align with image in vertical reader mode

plateaukao committed:
ui: remove xml style bookmark history context menu dialog

plateaukao committed:
ui: show bookmark context menu according to its long pressed item position

plateaukao committed:
ui: make history context menu dialog show up corresponding to its position

plateaukao committed:
ui: if it's bookmark folder, don't show open tab function in context menu

plateaukao committed:
feat: even if there's no second caption language setup, einkbro can still save current caption in youtube as epub

plateaukao committed:
fix: adblock releaseDebuggable flavor build issue

plateaukao committed:
fix: sometimes openai stream will stop but it's not failure

plateaukao committed:
feat: process digits in vertical read

plateaukao committed:
feat: support vertical english signle character or with a dot

plateaukao committed:
refactor: make vertical digit conersion more robust

plateaukao committed:
feat: optimize handling of ol list label in vertical reading mode

Jacek-Gawel committed:
Merge branch 'main' into feature/url_unification
